File: 1316.py
#1316

n = int(input())
result = 0

def groupcheck(word):
    check = set()
    a = word[0]
    check.add(a)
    for i in range(1, len(word)):
        if word[i] not in check:
            check.add(word[i])
            a = word[i]
            continue
        if word[i] != a and word[i] in check:
            return 0
        a = word[i]
    return 1

for i in range(n):
    result += groupcheck(str(input()))
print(result)

File: 1316.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Stack;

public class Main {
	public static void main(String[] args) {

		Scanner keyboard = new Scanner(System.in);

		int num = Integer.parseInt(keyboard.nextLine());
		
		int count = 0;

		for (int i = 0; i < num; ++i) {
			if(check(keyboard.nextLine())) count++;
		}
		System.out.println(count);
	}
	
	static boolean check(String word) {
		Stack<Character> stack = new Stack<>();
		ArrayList<Character> list = new ArrayList<Character>();
		
		for(int i = 0; i < word.length(); ++i) {
			if(stack.empty()) stack.push(word.charAt(i));
			else if (stack.peek() == word.charAt(i)) stack.push(word.charAt(i));
			else {
				list.add(stack.pop());
				while(!stack.empty())
					stack.pop();
			}
			Character checkChar = new Character(word.charAt(i));
			if (list.contains(checkChar)) return false;
		}
		return true;
	}
}
